### PR TITLE
feat(gitlab): add support for gitlab job rules needs

### PR DESCRIPTION
**Pull request by jacktantram**
`rules:needs` documentation can be found https://docs.gitlab.com/ee/ci/yaml/index.html#rulesneeds

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/api/gitlab.md [hunk 1]
```diff
@@ -2648,6 +2648,7 @@ const includeRule: gitlab.IncludeRule = { ... }
 | <code><a href="#projen.gitlab.IncludeRule.property.changes">changes</a></code> | <code>string[]</code> | *No description.* |
```
Comment (jacktantram) [inline docs/api/gitlab.md:2648]: In a separate pr I would be more than happy to update the description here.
Ideally with descriptions coming from https://docs.gitlab.com/ee/ci/yaml/index.html#gitlab.IncludeRule.property.if

Comment (mrgrain) [reply docs/api/gitlab.md:2648]: Would love that! 😍

**Review by mrgrain — APPROVED** [review]

**Comment by jacktantram** [comment]
Apologies for the back and forth on the build pipeline. 
I have now managed to get the check to run on the [fork](https://github.com/jacktantram/projen/actions/runs/9852095806/job/27200070268)
and is now passing.

Should be good to be retriggered now 👍

File: docs/api/gitlab.md (hunk 1 above)
Comment (jacktantram) [reply docs/api/gitlab.md:2648]: Okay great! I will push something up once this PR gets through 😄

**Review by mrgrain — APPROVED** [review]

**Pull request merged**